### PR TITLE
HHH-9186

### PR DESCRIPTION
**Pull request by ffeisst**
https://hibernate.atlassian.net/browse/HHH-9186

Added an unit test and bug fix for the issue.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/collection/internal/AbstractPersistentCollection.java.orig [hunk 1]
```diff
@@ -0,0 +1,1204 @@
+/*
```
Comment (brmeyer) [inline hibernate-core/src/main/java/org/hibernate/collection/internal/AbstractPersistentCollection.java.orig]: Hey @ffeisst, this file needs removed.

**Comment by ffeisst** [comment]
@brmeyer I have removed the file.

**Comment by vladmihalcea** [comment]
Thanks for the Pull Request. Did you sign the [CLA](https://cla.jboss.org/)?

**Comment by vladmihalcea** [comment]
Applied upstream, thanks

**Pull request closed without merging**